devnet-sdk/testing/testlib/validators/validators_test.go

```diff
@@ -113,6 +313,54 @@ func (m *mockChain) PendingNonceAt(ctx context.Context, address common.Address)
 func (m *mockChain) SupportsEIP(ctx context.Context, eip uint64) bool {
 	return true
 }
+func (m *mockChain) Config() (system.ChainConfig, error) {
+	if m.config == nil {
+		return nil, fmt.Errorf("chain config not implemented")
+	}
+	return m.config, nil
+}
+func (m *mockChain) BlockByNumber(ctx context.Context, number *big.Int) (*ethtypes.Block, error) {
+	time := uint64(1)
+	if m.config != nil {
+		time = m.config.currentTimestamp
+	}
+	header := &ethtypes.Header{
+		Time: time,
+	}
+	// Create a minimal block with just the timestamp
+	return ethtypes.NewBlockWithHeader(header), nil
+}
+
+// mockChainConfig implements the system.ChainConfig interface for testing
+type mockChainConfig struct {
+	activeForks      map[rollup.ForkName]uint64
+	currentTimestamp uint64
+	interopEnabled   bool
+}
+
+func (m *mockChainConfig) IsForkActivated(fork rollup.ForkName, timestamp uint64) bool {
+	activationTime, exists := m.activeForks[fork]
+	return exists && timestamp >= activationTime
+}
+
+func (m *mockChainConfig) CurrentFork(timestamp uint64) rollup.ForkName {
+	var currentFork rollup.ForkName
+	var highestActivationTime uint64
+
+	// Find the most recent fork that's active
+	for fork, activationTime := range m.activeForks {
+		if timestamp >= activationTime && (currentFork == "" || activationTime > highestActivationTime) {
+			currentFork = fork
+			highestActivationTime = activationTime
+		}
+	}
+
+	return currentFork
+}
+
+func (m *mockChainConfig) IsInteropEnabled(timestamp uint64) bool {
```

Comment:
No need this.

Reply:
Yup, this code is out of date because I am moving back to a ChainConfig approach.